examples/with-vite/src/main.tsx

```diff
@@ -2,36 +2,51 @@ import './polyfills';
 import './global.css';
 import '@rainbow-me/rainbowkit/styles.css';
 import { getDefaultWallets, RainbowKitProvider } from '@rainbow-me/rainbowkit';
-import { configureChains, createConfig, WagmiConfig } from 'wagmi';
-import { mainnet, polygon, optimism, arbitrum, base, zora } from 'wagmi/chains';
-import { publicProvider } from 'wagmi/providers/public';
+import { createConfig, http, WagmiProvider } from 'wagmi';
+import {
+  mainnet,
+  polygon,
+  optimism,
+  arbitrum,
+  base,
+  zora,
+  Chain,
+} from 'wagmi/chains';
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import App from './App';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 
-const { chains, publicClient } = configureChains(
-  [mainnet, polygon, optimism, arbitrum, base, zora],
-  [publicProvider()]
-);
+const chains = [mainnet, polygon, optimism, arbitrum, base, zora];
 
 const { connectors } = getDefaultWallets({
   appName: 'RainbowKit demo',
   projectId: 'YOUR_PROJECT_ID',
-  chains,
 });
 
 const wagmiConfig = createConfig({
-  autoConnect: true,
+  chains: chains as unknown as readonly [Chain, ...Chain[]],
```

Comment:
Going to fix this but `wagmi` uses tuple types as `readonly [Chain, ...Chain[]]`

Reply:
how about 

```ts
const chains = [mainnet, polygon, optimism, arbitrum, base, zora] as const;
```
and
```ts
export interface RainbowKitProviderProps {
  chains: ReadonlyArray<RainbowKitChain>;
  ...
```